### ISSUE TITLE
Add support for ROCm 2.4

### ISSUE THREAD

**Issue by andrerenard** [kotekan/kotekan#423]
Currently kotekan only works with ROCm 1.9-2.3, the changes listed below should get it working with ROCm 2.4.   However since the site is running 1.9 well, we don't plan to implement these changes for now, since they are not easily made backwards compatible. 

Notes from @kvand below:

…just to get it in writing (somewhere non-slack), I think the changes required to make kotekan work on the 2.4 driver are as below. I don’t think we should implement these, but should stick to the 1.9 driver for the time being, at least until there’s a very strong reason to switch up…

1) Force clang to use the v2 kernel format:

```
> diff --git a/lib/hsa/kernels/CMakeLists.txt b/lib/hsa/kernels/CMakeLists.txt
> index 2e803aca..a37b14f6 100644
> --- a/lib/hsa/kernels/CMakeLists.txt
> +++ b/lib/hsa/kernels/CMakeLists.txt
> @@ -44,7 +44,7 @@ FOREACH(infileName ${inISAFiles})
> 
>      # Custom command to build the hasco
>      ADD_CUSTOM_COMMAND(OUTPUT "${outfile}"
> -        COMMAND ${AMD_CLANG} -x assembler -target amdgcn--amdhsa -mcpu=fiji -c -o "${objectfile}" "${infile}"
> +        COMMAND ${AMD_CLANG} -x assembler -target amdgcn--amdhsa -mcpu=fiji -mno-code-object-v3 -c -o "${objectfile}" "${infile}"
>          COMMAND ${AMD_CLANG} -target amdgcn--amdhsa "${objectfile}" -o "${outfile}"
>          DEPENDS "${infile}" ${AMD_CLANG}
>          COMMENT "Generating hasco file from ${infile}: ${outfile}")
```

2) Add “.kd” to opencl kernel names, e.g.
```
> diff --git a/lib/hsa/hsaPresumKernel.cpp b/lib/hsa/hsaPresumKernel.cpp
> index 4f6b4cb8..ccff28b4 100644
> --- a/lib/hsa/hsaPresumKernel.cpp
> +++ b/lib/hsa/hsaPresumKernel.cpp
> @@ -9,7 +9,7 @@ hsaPresumKernel::hsaPresumKernel(Config& config, const string& unique_name,
>                                   bufferContainer& host_buffers, hsaDeviceInterface& device) :
>      hsaSubframeCommand(config, unique_name, host_buffers,
>                         //                                 device, "CHIME_presum","presum.hsaco") {
> -                       device, "CHIME_presum", "presum_opencl.hsaco") {
> +                       device, "CHIME_presum.kd", "presum_opencl.hsaco") {
>      command_type = gpuCommandType::KERNEL;
> 
>      _num_elements = config.get<int32_t>(unique_name, "num_elements");
```

**Comment by kvand** [comment]
Ran some verification tests, the two patches above do seem to fix things and result in correct results from ROCm 2.4. At a glance, it fixes timing instability on the FuryX nodes (not an issue on CHIME nodes) and leave performance more-or-less where it was. No major benefits I can see from making the jump.

The first seems to be down to bump in the code-object version from v2 to v3 in the latest ROCm. The second is from a change in the symbol naming convention in `llc`, the name remains the opencl name, but the symbol pointer to the kernel descriptor has the .kd appended. (I'm actually a little surprised it worked before, the symbol seems to have \@kd in the older versions.)

**Comment by james-s-willis** [comment]
I've created #435 to address these issues.

**Comment by james-s-willis** [comment]
Kotekan compiles assuming that the version of ROCm being used is 2.4 or greater. I've added a flag `USE_OLD_ROCM` to the CMake which will enable compilation for ROCm versions 2.3 or older.

**Comment by andrerenard** [comment]
This is resolved by #435
